### DISCUSSION TITLE
How to find existence of many-to-many children records that meet a criteria?

### DISCUSSION THREAD

**Discussion by nelsonjd** [adonisjs/core#3633] [Help]
I'm trying to find existence of certain Permissions on a single parent Role (many-to-many).
```
const roles = await Role
  .query()
  .preload('permissions')

this.role = roles.find(role => role.id === someid)

const exists = this.role.permissions.some(permission => permission.name === 'something')
```
This is the solution so far, which seems not optimal.
Is there a way to do this without loading all the roles and permissions and looping through them in Javascript? 
Thank you so much 🙏

**Comment by thetutlage** [comment]
If I get it right, you want to find if a role with a specific id has a given permission or not. If yes, you can use the following query.

```ts
const matchingRole = await Role.query().where('id', someId).whereHas('permissions', (subquery) => {
  subquery.where('name', 'something')
}).first()

if (!matchingRole) {
  // no match found
}
```
